### PR TITLE
Timestamp fix

### PR DESCRIPTION
**Pull request by Eugene-Roslikov-BQ**
Fix getTimestamp()
Fix getTime() and getDate() when calendar parameter is null

DateTimeDataConversionTests test category in test framework was used for testing.

Fix for the following behaviour of FlightSQL JDBC driver in compare to PostgreSQL JDBC driver:

getData() with null calendar return stored value minus 1 day. Should return just stored value.

getTime()  with null calendar return stored value minus 8 hours. Should return just stored value.

getTimestamp()  with default calendar or PST calendar return stored value plus 8 hours. Should return just stored value.

getTimestamp()  with UTC calendar return just stored value. Should return stored value minus 8 hours.

### CODE REVIEW

File: java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/utils/DateTimeUtils.java [hunk 1]
```diff
@@ -32,7 +33,7 @@ private DateTimeUtils() {
    */
   public static long applyCalendarOffset(long milliseconds, Calendar calendar) {
     if (calendar == null) {
-      return milliseconds;
+      return milliseconds - Calendar.getInstance(TimeZone.getDefault()).getTimeZone().getOffset(milliseconds);
```
Comment (vfraga) [inline java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/utils/DateTimeUtils.java:36]: Shouldn't we use UTC by default? I feel like using the system timezone is wrong (There's even a ticket to remove this behavior in the legacy driver).

**Review by vfraga — COMMENTED** [review]
That's my only concern. I think everything else is okay.

File: java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/utils/DateTimeUtils.java (hunk 1 above)
Comment (Eugene-Roslikov-BQ) [reply java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/utils/DateTimeUtils.java:36]: this is behaviour we have in  PostgreSQL jdbc driver.  Dremio jdbc driver just throw on null calendar.

**Comment by jduo** [comment]
> Fix getTimestamp() Fix getTime() and getDate() when calendar parameter is null

This isn't a sufficient description of this fix.
What's the issue? How do you reproduce it? How is it solved?

**Review by joyc-bq — APPROVED** [review]
looks good, though I didn't work on this code so look to rafael and others approval before merge

File: java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/calendar/ArrowFlightJdbcTimeStampVectorAccessorTest.java [hunk 2]
```diff
@@ -176,13 +176,12 @@ public void testShouldGetTimestampReturnValidTimestampWithCalendar() throws Exce
...
-      final Timestamp resultWithoutCalendar = accessor.getTimestamp(null);
       final Timestamp result = accessor.getTimestamp(calendar);
 
-      long offset = timeZone.getOffset(resultWithoutCalendar.getTime()) -
-          timeZoneForVector.getOffset(resultWithoutCalendar.getTime());
+      long offset = timeZone.getOffset(result.getTime()) -
+          timeZoneForVector.getOffset(result.getTime());
 
-      collector.checkThat(resultWithoutCalendar.getTime() - result.getTime(), is(offset));
+      collector.checkThat(result.getTime(), is(getTimestampForVector(currentRow).getTime() - offset ));
```
Comment (rafael-telles) [inline java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/calendar/ArrowFlightJdbcTimeStampVectorAccessorTest.java:184]:
```suggestion
      collector.checkThat(result.getTime(), is(getTimestampForVector(currentRow).getTime() - offset));
```
Lets just be sure that checkstyle doesn't complain about anything (run `mvn install` without additional flags)

**Pull request merged**